### PR TITLE
VAULT-22640: Add secret sync associations to precomputed queries

### PR DESCRIPTION
**Pull request by miagilepner**

### CODE REVIEW

File: vault/activity_log_test.go [hunk 1]
```diff
@@ -4274,13 +4274,15 @@ func TestActivityLog_processNewClients_delete(t *testing.T) {
 	mount := "mount"
 	namespace := "namespace"
 	clientID := "client-id"
-	run := func(t *testing.T, isNonEntity bool) {
+	run := func(t *testing.T, clientType string) {
 		t.Helper()
+		isNonEntity := clientType == entityActivityType || clientType == ACMEActivityType
```
Comment (hghaf099) [inline vault/activity_log_test.go:4279]: Should the first equality be `clientType == nonEntityTokenActivityType` instead?

Comment (hghaf099) [reply vault/activity_log_test.go:4279]: If that is true, it tells me that we could also add a test for the interaction of that field with the delete api.

File: vault/activity_log.go [hunk 2]
```diff
@@ -1976,49 +1977,94 @@ type (
...
+	if ok {
+		_, ok = byType[client.ClientID]
 		return ok
 	}
-	_, ok := p.Entities[client.ClientID]
-	return ok
+	return false
+}
+
+func (p *processCounts) toCountsRecord() *activity.CountsRecord {
```
Comment (hghaf099) [inline vault/activity_log.go:2034]: This is a nice helper function.

File: vault/activity_log.go [hunk 3]
```diff
@@ -2289,29 +2316,29 @@ func (a *ActivityLog) segmentToPrecomputedQuery(ctx context.Context, segmentTime
 		if segmentTime == opts.activePeriodEnd {
 			a.metrics.SetGaugeWithLabels(
```
Comment (hghaf099) [inline vault/activity_log.go:2317]: should a metric be added for secret sync assoc as well? Or is it gonna be in another PR?

**Review by hghaf099 — COMMENTED** [review]
Great work! Just had a few nit questions, comments.

File: vault/activity_log.go (hunk 3 above)
Comment (miagilepner) [reply vault/activity_log.go:2317]: that will be in a later PR

**Review by hghaf099 — APPROVED** [review]
LGTM!
